.github/workflows/test_on_push.yml

```diff
@@ -28,7 +28,7 @@ jobs:
       - name: Check style
         run: |
           python -m pip install pre-commit
-          pre-commit run ruff
+          pre-commit run --files $(git diff --name-only HEAD^)
```

Comment:
```suggestion
          git add .
          pre-commit run --files $(git diff --name-only HEAD^)
```

Reply:
Thanks for this @Saransh-cpp , 
to cross check this i added trailing whitespaces in a python file and from the terminal i gave these 2 commands , but they did not fail.

However I tried doing these steps:
1. Purposefully messed up a file
2. Committed that file 
3. Ran these commands 
`
changed_files=$(git diff --name-only HEAD^..HEAD)
pre-commit run --files $changed_files
`
4.This ran only on the changed file and failed.


Please let me know your thoughts on this. Thanks :D

Reply:
@Saransh-cpp thanks , explored on lint 
messed up a file in local and tried your method
Seems to be working fine.I agree we should let git and pre-commit handle their work themselves :D 
Please let me know if it looks good . 

<img width="677" alt="image" src="https://github.com/pybamm-team/PyBaMM/assets/32607282/8e28e6c6-0193-4b14-8619-6187cd32b992">